File: Assets/Scripts/misc/WaterFX/WaterShaderHelper.cs
using UnityEngine;

namespace Bootcamp.Misc.WaterFX
{
    [AddComponentMenu("Bootcamp/Misc/Water FX/Water Shader Helper")]
    public class WaterShaderHelper : MonoBehaviour
    {
        [SerializeField] Transform _lightDirection;
        private Material _material;

        private void Start()
        {
            _material = GetComponent<SkinnedMeshRenderer>().material;
        }

        public void Update()
        {
            _material.shader.maximumLOD = QualitySettings.GetQualityLevel() * 100;

            if (_lightDirection)
                _material.SetVector("_WorldLightDir", _lightDirection.forward);
            else
                _material.SetVector("_WorldLightDir", new Vector3(0.7f, 0.7f, 0.0f));
        }
    }
}

File: Assets/Scripts/Unity/WaterShaderHelper.js


var lightDir : Transform;

function Update () 
{
    var mat : Material = GetComponent.<Renderer>().material;
	mat.shader.maximumLOD = GameQualitySettings.water ? 600 : 300;
	if(lightDir)
		mat.SetVector("_WorldLightDir", lightDir.forward);
	else
		mat.SetVector("_WorldLightDir", Vector3(0.7,0.7,0.0));
}
